### ISSUE TITLE
Getting an error

### ISSUE THREAD

**Issue by msurguy** [gwygonik/SquiggleDrawInkscape#1]
I wasn't sure how to process an image so I copied a random image from Google search, then pasted into Inkscape, then with the image selected I went to Extensions -> Render -> SquiggleDraw

After I tried to turn on live preview or apply the filter, I got the following error:
![inkscape_pEDxbnfPKm](https://github.com/gwygonik/SquiggleDrawInkscape/assets/585833/8f05a27c-cc42-4ffb-b144-0b43dfcfab98)

Running on Windows 10, 64 bit, latest Inkscape version (1.3+)

**Comment by gwygonik** [comment]
Great! (Seriously, I was hoping for some testing, especially on Windows). Can you reproduce and provide a screencap of the extension dialog? I've seen Inkscape not limiting the values to their bounds if you use the up and down arrows, but it does if you drag the slider. If that is the issue here, I can happily code for that, though that would be an Inkscape bug. If that's NOT the issue in this case, then I'll have to get some more details :-)

**Comment by gwygonik** [comment]
Ok! I found there _was_ a bug with the divisors in the code, but also can get Inkscape to go beyond the bounds (though can't reproduce consistently). I updated the code to fix this issue, and also added an additional level of amplitude.

Please pull the latest (v1.0.1) and see if you still get the error.

**Comment by gwygonik** [comment]
Additional note: there's not a way to have Inkscape alert the user to have an image selected before actually applying the extension. I've added text that says to have an image selected before running the extension, but you can't see that before running the extension! D'oh! The only(?) way to really explain that is probably here in the README file, but that's also somewhat annoying. Not sure what else to do about this. :-\

**Comment by oskay** [comment]
I'd suggest that if an image isn't selected, that you present the user with an error message, asking them to select an image first.

**Comment by gwygonik** [comment]
> I'd suggest that if an image isn't selected, that you present the user with an error message, asking them to select an image first.

The extension does do that, but only in the .py file which run _after_ the UI is presented and options selected. The .inx file (the UI) doesn't appear to have any way to programmatically do any conditional/logic processing ☹️

**Comment by oskay** [comment]
Giving an error if the right thing isn't selected is normal behavior in Inkscape extensions, so long as it's handled gracefully to avoid a traceback for example.

**Comment by gwygonik** [comment]
Agreed, though it would be better (IMO) if the user could be alerted before getting all the way to trying to apply the extension before seeing an error message. If the UI could have prerequisites (e.g.: requires_selection=true), then an upfront alert would alleviate confusion/frustration. But that's neither here nor there for this extension 😄 I think the inkex processor would need to have a significant update to handle this at a core level.

**Comment by oskay** [comment]
Inkex doesn't run until the extension does; it would have to be a core change in Inkscape to enable something like that. But, tons of extensions already work this way so keeping the same use pattern will be natural and correct to most Inkscape users.

**Comment by gwygonik** [comment]
The extension will show an error if you try to apply it without an image (or anything) selected.

**Comment by msurguy** [comment]
I downloaded the latest version and everything's working correctly now! 
One thing that could be better is to show floating point numbers in the sliders, currently they show up as integers even though the values are fractional.

**Comment by gwygonik** [comment]
> One thing that could be better is to show floating point numbers in the sliders, currently they show up as integers even though the values are fractional.

Wait what? 🧐  All parameters from the UI are showing as integers because they are set to be INTs, and are used as such. They are used in calculations for other variables, but always expected to be INTs. If there is somewhere you're seeing a param being a float, please let me know! (I'd seriously hope that Inkscape is using the explicitly-defined parameter types correctly...) [re-opening just to allow comments]

**Comment by msurguy** [comment]
@gwygonik perhaps this illustrates the issue. Essentially I have amplitude slider tell me it's value is 2, but I guess there are a few levels of 2? 

![chrome_jMEnLsxS8p](https://github.com/gwygonik/SquiggleDrawInkscape/assets/585833/28350b4d-f2d0-4fd4-92e1-706e502371c0)

**Comment by gwygonik** [comment]
I did some debugging and can't get any of the parameters to come through as floats. However, I added code to force them all to be ints (after flooring them) and would love for you to try it again. 

Also, was there anything different in the way you added the image? Or the size of the image between passes? I wonder if there's a different issue related to the way I make/get the pixel brightness... 

Thanks!!
